### PR TITLE
Allow setting default build target

### PR DESCRIPTION
**Pull request by zappolowski**
This implements #368 

This feature allows to specify the default target to build either in `Cross.toml`

```toml
[build]
target = "aarch64-unknown-linux-gnu"
```
or by setting `CROSS_BUILD_TARGET` to avoid the need to always specify `--target`.

I've opted for `build.target` as this resembles the configuration of [cargo](https://doc.rust-lang.org/cargo/reference/config.html#buildtarget).

### CODE REVIEW

**Review by Emilgardis — DISMISSED** [review]
Looks good! Cant merge right now due to CI being incomplete

**Comment by zappolowski** [comment]
> Cant merge right now due to CI being incomplete

Is some action required on my site to complete the CI?

**Comment by Emilgardis** [comment]
No, see #609 <3

**Comment by reitermarkus** [comment]
Does this need to be in `Cross.toml`? Can't this be in `.cargo/config.toml` and we automatically detect it?

**Comment by zappolowski** [comment]
> Does this need to be in Cross.toml? Can't this be in .cargo/config.toml and we automatically detect it?

I've opted for using `Cross.toml` in the first place as it's a configuration made for `cross` and shouldn't affect `cargo`. Another issue was/is that according to [the documentation](https://doc.rust-lang.org/cargo/reference/config.html) the actual value can be defined in several places and duplicating the resolution implemented by `cargo` (and keeping it up to date) seemed unnecessary complex.

My proposal would be to postpone this feature until [cargo config](https://github.com/rust-lang/cargo/issues/9301) is stabilized as this should make it possible to just parse the output of `cargo config get --format toml build.target` and use this without the hassle mentioned above.

**Comment by reitermarkus** [comment]
> My proposal would be to postpone this feature until https://github.com/rust-lang/cargo/issues/9301 is stabilized

Sounds fair.

Ideally I would like to get rid of `Cross.toml` eventually and use `package.metadata.cross` in `Cargo.toml` to store `cross` configuration.

**Review by otavio — APPROVED** [review]

**Comment by otavio** [comment]
I don't see why this should be postponed as this is very useful and a valid use case. Once the cargo feature is merged this can be reworked in a new release.

**Comment by Emilgardis** [comment]
can you please rebase this again so we can get this merged? There's been some changes with #670

**Comment by Emilgardis** [comment]
I've renamed the option to `default-target` to better convey what it does

**Review by Emilgardis — APPROVED** [review]
bors r+

thanks!

**Pull request merged**
